### PR TITLE
chore(sdks): Update format to allow Go SDK generation

### PR DESCRIPTION
**Pull request by steve-gr4vy**
# Description

These changes are required to 'help' the go openapi generator create correct working sdk.  I have also tested the changes against the node sdk.

## Checklist

- [x ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own changes
- [x ] I have run `yarn test` to make sure my changes pass all linters
- [x ] I have pulled the latest changes from the upstream `main` branch

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

### CODE REVIEW

File: reference/resources/card_rule_number_condition.yaml [hunk 1]
```diff
@@ -32,13 +32,6 @@ properties:
     type: string
     description: The comparison to make to `value` property.
     example: ">="
-    enum:
-      - "=="
-      - "!="
-      - "<="
-      - ">="
-      - ">"
-      - "<"
```
Comment (cbetta) [inline reference/resources/card_rule_number_condition.yaml:35]: Why did we remove this?

File: reference/resources/payment_service_definition.yaml [hunk 2]
```diff
@@ -63,10 +63,6 @@ properties:
           example: text
           description: |-
             Defines the type of input that needs to be rendered for this field.
-          enum:
-            - text
-            - multiline
-            - number
```
Comment (cbetta) [inline reference/resources/payment_service_definition.yaml:69]: Likewise, why did we remove these?

**Review by cbetta — APPROVED** [review]

**Review by cbetta — CHANGES_REQUESTED** [review]
Bit unclear why we removed some enums.

File: reference/resources/card_rule_number_condition.yaml (hunk 1 above)
Comment (steve-gr4vy) [reply reference/resources/card_rule_number_condition.yaml:35]: The Go sdk generator was creating duplicate types that caused compile errors - e.g. 
`
api/gr4vy.gen.go:165:2: CardRuleNumberConditionOperator is not a type
api/gr4vy.gen.go:167:2: CardRuleNumberConditionOperator is not a type
api/gr4vy.gen.go:169:2: CardRuleNumberConditionOperator is not a type
api/gr4vy.gen.go:171:2: CardRuleNumberConditionOperator is not a type
api/gr4vy.gen.go:173:2: CardRuleNumberConditionOperator is not a type
api/gr4vy.gen.go:474:2: PaymentServiceDefinitionFieldsFormatMultiline redeclared in this block
	previous declaration at api/gr4vy.gen.go:465:87
api/gr4vy.gen.go:476:2: PaymentServiceDefinitionFieldsFormatNumber redeclared in this block
	previous declaration at api/gr4vy.gen.go:467:84
api/gr4vy.gen.go:478:2: PaymentServiceDefinitionFieldsFormatText redeclared in this block
	previous declaration at api/gr4vy.gen.go:469:82
api/gr4vy.gen.go:954:6: CardRuleNumberConditionOperator redeclared in this block
	previous declaration at api/gr4vy.gen.go:163:68
api/gr4vy.gen.go:1656:6: PaymentServiceDefinitionFieldsFormat redeclared in this block
	previous declaration at api/gr4vy.gen.go:1653:6
`

**Review by cbetta — CHANGES_REQUESTED** [review]
Sorry but not going to be able to approve this. We can't just go and strip some of the enums for an SDK. We either need to update the SDK Spec build (https://github.com/gr4vy/gr4vy-openapi/blob/main/scripts/build-sdk-spec.js) or clean up the spec after it's been downloaded for use by the Go generator.

**Review by cbetta — APPROVED** [review]

**Pull request merged**